src/user.js

```diff
@@ -0,0 +1,169 @@
+import getAuthenticationDetails from './helpers/getAuthenticationDetails';
+import getCognitoUser from './helpers/getCognitoUser';
+import getUserPool from './helpers/getUserPool';
+import getCognitoUserAttributes from './helpers/getCognitoUserAttributes';
+
+const user = class User {
+  constructor(awsCognito) {
+    this.awsCognito = awsCognito;
+  }
+
+  auth(username, password, userPoolId, clientId) {
+    const authenticationDetails = getAuthenticationDetails(username, password,
+      this.awsCognito.AuthenticationDetails);
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.authenticateUser(authenticationDetails, {
+        onSuccess(result) {
+          resolve(result);
+        },
+        onFailure(err) {
+          reject(err);
+        },
+      });
+    });
+  }
+
+  create(username, password, userPoolId, clientId, attributes) {
+    const userPool = getUserPool(userPoolId, clientId, this.awsCognito.CognitoUserPool);
+    const cognitoUserAttributes = getCognitoUserAttributes(attributes,
+      this.awsCognito.CognitoUserAttribute);
+
+    return new Promise((resolve, reject) => {
+      userPool.signUp(username, password, cognitoUserAttributes, null, (err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  confirmRegistration(username, confirmationCode, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.confirmRegistration(confirmationCode, true, (err, result) => {
+        if (err) {
```

Comment:
I've created a generic promise callback function for this now - removes the duplication quite a bit.